File: DSA/HR/diagonal_difference.java
import java.util.List;

class Result {

    public static int diagonalDifference(List<List<Integer>> arr) {
        int leftDiagonal = 0, rightDiagonal = 0;
        for (int i = 0; i < arr.size(); i++) {
            leftDiagonal += arr.get(i).get(i);
            rightDiagonal += arr.get(i).get(arr.size() - 1 - i);
        }

        return Math.abs(leftDiagonal - rightDiagonal);
    }

}

File: DSA/HR/diagonal_difference.cpp
#include <iostream>
#include <vector>
#include <cmath>

using namespace std;

int diagonalDifference(vector<vector<int>> arr)
{
    int n = arr.size();
    int primary_diagonal_sum = 0;
    int secondary_diagonal_sum = 0;

    for (int i = 0; i < n; i++)
    {
        primary_diagonal_sum += arr[i][i];
        secondary_diagonal_sum += arr[i][n - 1 - i];
    }

    int difference = abs(primary_diagonal_sum - secondary_diagonal_sum);

    return difference;
}